libsolidity/interface/CompilerStack.cpp

```diff
@@ -2005,6 +2016,27 @@ experimental::Analysis const& CompilerStack::experimentalAnalysis() const
 
 void CompilerStack::reportUnimplementedFeatureError(UnimplementedFeatureError const& _error)
 {
-	solAssert(_error.comment(), "Unimplemented feature errors must include a message for the user");
+	solAssert(_error.comment(), "Errors must include a message for the user.");
+
 	m_errorReporter.unimplementedFeatureError(1834_error, _error.sourceLocation(), *_error.comment());
 }
+
+void CompilerStack::reportIRPostAnalysisError(Error const* _error)
+{
+	solAssert(_error);
+	solAssert(_error->comment(), "Errors must include a message for the user.");
+	solAssert(!_error->secondarySourceLocation());
+
+	// Do not report Yul warnings and infos. These are only reported in pure Yul compilation.
```

Comment:
Is this comment correct? Shouldn't it say "do not report Yul errors" instead? Because you check if it is an error below and only avoid reporting it in that case, no?

Reply:
Well, the comment is correct. The intention was exactly what the comment says. The code totally isn't though :P It's missing a negation. 

Thanks for finding that. Another bug that slipped through because we currently don't have a way to cover it with a test case. This code will actually never run - not until we actually introduce some `CodeGenerationError` during Yul->EVM transform.